### ISSUE TITLE
"unparsed tail left after parsing float64 "

### ISSUE THREAD

**Issue by w6m6** [VictoriaMetrics/VictoriaMetrics#924]
**Describe the bug**
a error message where thrown using vmagent to get the metrics of  apache pulsar
`{"ts":"2020-11-24T14:02:27.711Z","level":"error","caller":"VictoriaMetrics/lib/protoparser/prometheus/parser.go:206","msg":"error when scraping \"http://172.21.216.43:8000/metrics\" from job \"pulsar-bookie\" with labels {catalog=\"pulsar\", dc=\"wg\", instance=\"172.21.216.43:8000\", ip=\"172.21.216.43\", job=\"pulsar-bookie\", rapp=\"platform\", rservice=\"bookie\", rstatus=\"active\", src=\"wg\"}: cannot unmarshal Prometheus line \"bookie_THREAD_RUNTIME{success=\\\"true\\\",quantile=\\\"1.0\\\"} -Infinity\": cannot parse value \"-Infinity\": unparsed tail left after parsing float64 from \"-Infinity\": \"Infinity\""}`
the metrics of apache pulsar listed as below.
`bookie_bookie_zk_get_data{success="false",quantile="0.5"}` NaN
bookie_bookie_zk_get_data{success="false",quantile="0.75"} NaN
bookie_bookie_zk_get_data{success="false",quantile="0.95"} NaN
bookie_bookie_zk_get_data{success="false",quantile="0.99"} NaN
bookie_bookie_zk_get_data{success="false",quantile="0.999"} NaN
bookie_bookie_zk_get_data{success="false",quantile="0.9999"} NaN
bookie_bookie_zk_get_data{success="false",quantile="1.0"} -Infinity
`
**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
vmagent-20201102-004420-tags-v1.45.0-0-gd396c265a
```

**Comment by valyala** [comment]
The bug should be fixed in the commit ae04378424c9171af06ecd3e7af91bebebe771cf . @w6m6 , could you build VictoriaMetrics according to [these docs](https://victoriametrics.github.io/#how-to-build-from-sources) and verify whether the bug has been fixed there?

**Comment by valyala** [comment]
The bugfix has been included in [v1.48.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this issue as fixed. @w6m6 , feel free re-opening it if it isn't fully fixed.

**Comment by w6m6** [comment]
@valyala confirmed that it's fixed in 1.48.0, thx a lot.
